kroxylicious-systemtests/src/test/java/io/kroxylicious/systemtests/NonJVMClientsST.java

```diff
@@ -0,0 +1,113 @@
+import io.kroxylicious.systemtests.steps.KafkaSteps;
+import io.kroxylicious.systemtests.steps.KroxyliciousSteps;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaNodePoolTemplates;
+import io.kroxylicious.systemtests.templates.strimzi.KafkaTemplates;
+
+import static io.kroxylicious.systemtests.k8s.KubeClusterResource.kubeClient;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * The Kroxylicious system tests.
+ */
+@ExtendWith(KroxyliciousExtension.class)
+class NonJVMClientsST extends AbstractST {
+    private static final Logger LOGGER = LoggerFactory.getLogger(NonJVMClientsST.class);
+    private static Kroxylicious kroxylicious;
+    private final String clusterName = "my-cluster";
+    protected static final String BROKER_NODE_NAME = "kafka";
+    private static final String MESSAGE = "Message";
+
+    /**
+     * Produce and consume message.
+     *
+     * @param namespace the namespace
+     */
+    @Test
+    void kCatClients(String namespace) {
+        int numberOfMessages = 1;
+        String expectedMessage = MESSAGE;
+
+        // start Kroxylicious
+        LOGGER.info("Given Kroxylicious in {} namespace with {} replicas", namespace, 1);
+        kroxylicious = new Kroxylicious(namespace);
+        kroxylicious.deployPortPerBrokerPlainWithNoFilters(clusterName, 1);
+        String bootstrap = kroxylicious.getBootstrap();
+
+        LOGGER.info("And a kafka Topic named {}", topicName);
+        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 2);
+
+        LOGGER.info("When {} messages '{}' are sent to the topic '{}'", numberOfMessages, MESSAGE, topicName);
+        KroxyliciousSteps.produceMessagesWithKcat(namespace, topicName, bootstrap, MESSAGE, numberOfMessages);
+
+        LOGGER.info("Then the {} messages are consumed", numberOfMessages);
+        String result = KroxyliciousSteps.consumeMessagesWithKcat(namespace, topicName, bootstrap, expectedMessage, Duration.ofMinutes(2));
+        LOGGER.info("Received: " + result);
+
+        assertThat(result).withFailMessage("expected message have not been received!").contains(expectedMessage);
+    }
+
+    @Test
+    void kafkaGoClients(String namespace) {
+        int numberOfMessages = 1;
+        String expectedMessage = MESSAGE + "-" + (numberOfMessages);
+
+        // start Kroxylicious
+        LOGGER.info("Given Kroxylicious in {} namespace with {} replicas", namespace, 1);
+        kroxylicious = new Kroxylicious(namespace);
+        kroxylicious.deployPortPerBrokerPlainWithNoFilters(clusterName, 1);
+        String bootstrap = kroxylicious.getBootstrap();
+
+        LOGGER.info("And a kafka Topic named {}", topicName);
+        KafkaSteps.createTopic(namespace, topicName, bootstrap, 1, 2);
+
+        LOGGER.info("When the messages '{}' are sent to the topic '{}'", MESSAGE, topicName);
+        KroxyliciousSteps.produceMessagesWithKafkaGo(namespace, topicName, bootstrap);
```

Comment:
I found this surprising that we log that it's sending MESSAGE but message isn't passed down to the producer. The [producer](https://github.com/ppatierno/kafka-go-examples/blob/master/segmentio/producer/main.go) is periodically generating messages like `Message-1` with an incrementing counter. Would be worth adding a comment.

I'd also support switching to `kaf` like Keith suggests so we are using more up to date versions of the go libs and can control the produced messages from the test code.

Reply:
Moved to `kaf`